src/main/java/org/opentripplanner/graph_builder/impl/shapefile/NihShapefileStreetGraphBuilderImpl.java

```diff
@@ -349,10 +391,94 @@ public void buildGraph(Graph graph, HashMap<Class<?>, Object> extra) {
         }
     }
 
+    private EnumMap<NihOption, ArrayList<StringAttributeFeatureConverter>> getFieldConverters() {
+        // if have R/L columns, have list [R, L]
+        EnumMap<NihOption, ArrayList<StringAttributeFeatureConverter>> converters = new EnumMap(NihOption.class);
+
+        final String LEFT_PREFIX = "L_";
+        final String RIGHT_PREFIX = "R_";
+
+        for (NihOption option : NIH_OPTIONS) {
+            ArrayList<StringAttributeFeatureConverter> fields = new ArrayList();
+            if (option.hasLeftRight()) {
+
+                if (option.equals(NihOption.TRAFFIC)) {
+                    continue; // TODO: will this field go into the shapefile eventually? skip it for now.
+                }
+
+                // make two converters, one for each side
+                fields.add(new StringAttributeFeatureConverter(RIGHT_PREFIX + option.getFieldName(), ""));
+                fields.add(new StringAttributeFeatureConverter(LEFT_PREFIX + option.getFieldName(), ""));
+            } else {
+                fields.add(new StringAttributeFeatureConverter(option.getFieldName(), ""));
+            }
+            converters.put(option, fields);
+        }
+
+        return converters;
+    }
+
+    private void setProperties(SimpleFeature feature, Set<StreetEdge> edges, boolean isRightSide) {
+
+        if (isRightSide) {
+            LOG.info("Going to set properties on right edges...");
+        } else {
+            LOG.info("Going to set properties on left edges...");
+        }
+
+        // build set of converted values
+        OptionSet<NihOption> nihOptions = new OptionSet(NihOption.class);
+
+        for (NihOption option : NIH_OPTIONS) {
+            String val = "";
+
+            if (option.equals(NihOption.TRAFFIC)) {
+                // TODO: will this column be added to the shapefile?
+                continue;
+            }
+
+            if (!option.hasLeftRight() || isRightSide) {
+                val = optionConverters.get(option).get(0).convert(feature);
+            } else {
+                // left side is second in list
+                val = optionConverters.get(option).get(1).convert(feature);
+            }
+
+            if (val.isEmpty()) {
+                LOG.warn("Skipping setting an empty value for option {}.  Was this intentional?", option.getFieldName());
+            } else {
+
+                // TODO: some better way to correct for left/right values encoded in a single column?
+                if (val.equals("Right")) {
+                    if (isRightSide) {
+                        val = "Both";
```

Comment:
Ah! I get it! 

Might be worth then placing a comment detailing this at the NihOptions class level, so newcomers are aware of this oddity?